### PR TITLE
Fix internal links in demo>documentation

### PR DESCRIPTION
**Pull request by Pessimistress**
- Generic url rewrite
- Support in-page jump links

### CODE REVIEW

File: demo/src/components/page.js [hunk 1]
```diff
@@ -88,16 +88,29 @@ class Page extends Component {
     );
   }
 
+  @autobind _updateQueryString(queryString) {
```
Comment (ibgreen) [inline demo/src/components/page.js]: Doc header...

File: demo/src/components/page.js [hunk 2]
```diff
@@ -88,16 +88,29 @@ class Page extends Component {
...
 
     let child;
 
     if (content.demo) {
       child = this._renderDemo(content.demo, content.code);
     } else if (typeof content === 'string') {
-      child = <MarkdownPage content={contents[content]} renderDemo={this._renderDemo} />;
+      child = (<MarkdownPage content={contents[content]}
+        query={query}
+        updateQuery={this._updateQueryString}
```
Comment (ibgreen) [inline demo/src/components/page.js]: Use same name for prop and function?

File: demo/src/components/markdown-page.js [hunk 3]
```diff
@@ -40,38 +42,123 @@ marked.setOptions({
...
+        anchorPositions: null
+      });
+    }
+  }
+
+  componentDidUpdate() {
+    this._jumpTo(this.props.query.section);
+  }
+
+  _getScrollPosition(object) {
```
Comment (ibgreen) [inline demo/src/components/markdown-page.js]: Doc header...

File: demo/src/components/markdown-page.js [hunk 4]
```diff
@@ -1,25 +1,27 @@
...
-  '/docs/custom-layers.md#layerencodepickingcolorindex--number': false,
-  '/docs/layers/arc-layer.md': '#/layers/core-layers/arc-layer',
-  '/docs/layers/line-layer.md': '#/layers/core-layers/line-layer',
-  '/docs/layers/geojson-layer.md': '#/layers/core-layers/geojson-layer',
-  '/docs/layers/scatterplot-layer.md': '#/layers/core-layers/scatterplot-layer',
-  '/docs/64-bits.md': '#/documentation/advanced-topics/64-layers'
-};
+const urlRewrites = [
+  {
+    test: /^\/(docs\/.+?\.md)(#.*)?$/,
```
Comment (shaojingli) [inline demo/src/components/markdown-page.js]: oh... regex... * _ * I would always want a plain english explanation for what a regex is doing...

**Pull request merged**